### PR TITLE
Navigation: melodic-devel

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
This works with the updated navigation stack on the melodic-devel branch. To work with HERO, hero_bringup and hero_bridge need be have their `fix/tmc20` branches merged. As it is too much work to get this to work with the older releases.

Merge together with https://github.com/tue-robotics/hero_bringup/pull/57, https://github.com/tue-robotics/hero_bridge/pull/16, https://github.com/tue-robotics/robot_launch_files/pull/59, https://github.com/tue-robotics/ed_navigation/pull/24 and https::/github.com/tue-robotics/navigation default branch changed to `melodic-devel`

### CODE REVIEW

File: global_planner_plugins/a_star_planner/a_star_planner_gpp.cpp [hunk 1]
```diff
@@ -12,49 +17,51 @@ PLUGINLIB_EXPORT_CLASS(cb_global_planner::AStarPlannerGPP, cb_global_planner::Gl
...
 
 // ----------------------------------------------------------------------------------------------------
 
-AStarPlannerGPP::AStarPlannerGPP() : global_costmap_ros_(NULL),  planner_(NULL) {}
+AStarPlannerGPP::AStarPlannerGPP() : global_costmap_ros_(nullptr), planner_(nullptr), tf_(nullptr)
+{
+}
 
-void AStarPlannerGPP::initialize(std::string name, tf::TransformListener* tf, costmap_2d::Costmap2DROS* global_costmap_ros)
+void AStarPlannerGPP::initialize(std::string /*name*/, tf2_ros::Buffer* tf, costmap_2d::Costmap2DROS* global_costmap_ros)
```
Comment (PetervDooren) [inline global_planner_plugins/a_star_planner/a_star_planner_gpp.cpp:57]: Why is name commented out? does this not break stuff?

**Review by PetervDooren — COMMENTED** [review]
I think I prefer "position_constraint" over "pc". That makes it easier for someone to understand the code and it doesn't cost us anything.

I must admit some things went a little over my head. It looks good to me now, but I'll read up on tf2 and give it another look over.

**Comment by PetervDooren** [comment]
Also check why CI is failing

File: global_planner_plugins/a_star_planner/a_star_planner_gpp.cpp (hunk 1 above)
Comment (MatthijsBurgh) [reply global_planner_plugins/a_star_planner/a_star_planner_gpp.cpp:57]: No, this is common to do when a argument is not used, but it is not possible to change the API.

**Comment by MatthijsBurgh** [comment]
PR CI are allowed to fail. Of course not because of all reasons. But if a PR depends on other PRs, the PR CI will fail, because in the PR CI, the current repo is checked out in a state which represents the state after this PR being merged. All other repos are therefore checked out to their default branch. So therefore stuff breaks.

This PR has a code dependency on the navigation stack and therefore it fails if the navigation is checked out to the old version.

**Review by PetervDooren — APPROVED** [review]

**Pull request merged**